frontend/src/components/nav-link.tsx

```diff
@@ -1,15 +1,15 @@
-import { Link, type LinkProps } from "@tanstack/react-router";
-import React from "react";
+import { Link } from "@tanstack/react-router";
+import { forwardRef } from "react";
+
+import type { LinkProps } from "@tanstack/react-router";
+import type { ElementRef } from "react";
```

Comment:
```suggestion
import * as React from "react";
```

`import * as React form 'react'` is a choice to distinguish between React hooks and custom hooks